ops/charm.py

```diff
@@ -488,6 +487,19 @@ def from_yaml(
         return cls(meta, raw_actions)
 
 
+class RelationRole(enum.Enum):
+    peer = 'peer'
+    requires = 'requires'
+    provides = 'provides'
+
+    def is_peer(self) -> bool:
```

Comment:
This is handy. But:

- totally unexpected
- what is the benefit of avoiding to import charm? is importing charm a performance problem? or mixing up everything? probably we should consider having a "types" module
- the "self is" idiom is weird: enums are normally compared by equality, why not doing that here?

Reply:
> totally unexpected

how so?

> what is the benefit of avoiding to import charm? is importing charm a performance problem? or mixing up everything? probably we should consider having a "types" module

the current code has
```python
            if relation_meta.role == 'peers':
```
this would become
```python
            if relation_meta.role == ops.charm.RelationRole.peer:
```
which I felt was too awkward, that you'd have to import things from `charm` just for this. Even if you import just what you need, it's not a lot better,
```python
            if relation_meta.role == RelationRole.peer:
```


I found
```python
            if relation_meta.role.is_peer():
```
to be an improvement: it reads as exactly what you want to check for.

Reply:
> the "self is" idiom is weird: enums are normally compared by equality, why not doing that here?

I mean, we can, but the documentation for `enum` mentions identity explicitly and with examples. Where do you get the 'normally compared by equality' from?

Reply:
[the docs in question](https://docs.python.org/3/library/enum.html#comparisons) FWIW.

Reply:
I think you're right.